packages/sdk-v2/src/sdk/ipfs/sdk.ts

```diff
@@ -0,0 +1,131 @@
+import { isString, sortBy } from "lodash";
+import { CID } from "multiformats/cid";
+
+import { fallback, withTemporaryFile } from "../../utils";
+
+interface RequestOptions {
+  headers?: any;
+  body?: any;
+  mode?: any;
+}
+
+// eslint-disable-next-line
+export const toV1 = cid => CID.parse(cid).toV1().toString();
+//todo: move to utils
+export const cidRegexp = /^[\w\d]+$/i;
+
+// checks is string a valid CID. it should be at least 40 chars length and contrain only letters & numbers
+export const isValidCID = source => isString(source) && source.length >= 40 && cidRegexp.test(source);
+
+export type IPFSUrls = { ipfsGateways: string; ipfsUploadGateway: string };
+
+export const defaulIPFS: IPFSUrls = {
+  ipfsGateways:
+    "https://{cid}.ipfs.nftstorage.link,https://cloudflare-ipfs.com/ipfs/{cid},https://ipfs.io/ipfs/{cid},https://{cid}.ipfs.dweb.link",
+  ipfsUploadGateway: "https://ipfsgateway.goodworker.workers.dev"
+};
+
+export class IpfsStorage {
+  client: any;
+  gateways: any;
+
+  constructor(ipfsUrls?: IPFSUrls) {
+    const { ipfsGateways, ipfsUploadGateway } = { ...defaulIPFS, ...ipfsUrls };
+
+    // add tpyes
+    this.client = {
+      request: async (method, url, options: RequestOptions = {}) => {
```

Comment:
just the enhancement - if you want to keep it let's make it separate util could be used elsewhere

```typescript
class HttpClient {
  constructor(private baseUrl) {}

  async request(method, url, options: RequestOptions = {}) {
         const updatedOptions = {
           ...options,
           method: method,
           headers: {
             ...options?.headers
           }
         };

         return fetch(this.baseUrl + url, updatedOptions);
       }

       async fetchBlob (url) {
         const updatedOptions = {
           method: "GET"
         };

         const response = await fetch(url, { method: "GET" });

         return response.blob();
       }
}


// ...

this.client = new HttpClient(ipfsUploadGateway)
```